File: test_ws.lua
-- WebSocket Test Client for Volt API

local function sendNotify(title, text)
    game:GetService("StarterGui"):SetCore("SendNotification", {
        Title = title,
        Text = text,
        Duration = 5
    })
end

print("[TEST CLIENT] Connecting to WebSocket...")
local ws = WebSocket.connect("ws://157.15.40.37:5555")

ws.OnMessage:Connect(function(message)
    print("[TEST CLIENT] Received:", message)
    if message == "pong" then
        sendNotify("WebSocket Test", "Received PONG from server!")
    end
end)

ws.OnClose:Connect(function()
    print("[TEST CLIENT] Connection closed")
    sendNotify("WebSocket Test", "Connection closed.")
end)

task.wait(2)
print("[TEST CLIENT] Sending PING...")
ws:Send("ping")

-- Keep script alive for the test
task.wait(10)
print("[TEST CLIENT] Closing connection...")
ws:Close()

File: test_ws.js
const WebSocket = require('ws');

const wss = new WebSocket.Server({ port: 5555 });

console.log('[TEST SERVER] WebSocket server started on ws://157.15.40.37:5555');

wss.on('connection', (ws) => {
    console.log('[TEST SERVER] Client connected');
    
    ws.on('message', (message) => {
        const msg = message.toString();
        console.log(`[TEST SERVER] Received: ${msg}`);
        
        if (msg === 'ping') {
            console.log('[TEST SERVER] Sending: pong');
            ws.send('pong');
        }
    });
    
    ws.on('close', () => {
        console.log('[TEST SERVER] Client disconnected');
    });
});
